ecs/src/nox/thread/Pool.tpp

```diff
@@ -0,0 +1,103 @@
+#include <algorithm>
+#include <chrono>
+
+template<template<class> class QueueType>
+nox::thread::Pool<QueueType>::Pool(std::size_t threadCount)
+    : threads(threadCount)
+{
+    auto workerFunc = [this]() -> void
+    {
+        while (std::atomic_load_explicit(&this->shouldContinue, std::memory_order_seq_cst))
+        {
+            std::unique_lock<std::mutex> lock(this->cvMutex);
+            Task task;
+            
+            // Continue waiting if: No tasks are popped from the queue, and we should continue.
```

Comment:
Should change logic of comment to align with how the logic happens in the code. I suggest "A task is popped from the queue, or we should not continue".
Changing the logic of the code to match the comment could also be a solution.